components/composited/PasswordForm/PasswordForm.tsx

```diff
@@ -0,0 +1,79 @@
+import { FormEventHandler, Fragment, useRef, useState } from 'react';
+
+import { PasswordChangeFields, PasswordChangeSch } from '../../../validators/PasswordChange';
+import { Button } from '../../generic/Button/Button';
+import { Input } from '../../generic/Input/Input';
+import { ErrorMessage } from '../../generic/ErrorMessage/ErrorMessage';
+import { WarningMessage } from '../../generic/WarningMessage/WarningMessage';
+
+type PasswordFormProps = {
+  onSubmit: (fields: PasswordChangeFields) => void;
+};
+
+export const PasswordForm = ({ onSubmit }: PasswordFormProps) => {
+  const formRef = useRef<HTMLFormElement>(null);
+  const [inputErrorMessage, setInputErrorMessage] = useState(false);
+  const [differentPasswords, setDifferentPasswords] = useState(false);
```

Comment:
We might need to Proxy that to the server, if supabase doesn't support that option.